### PR TITLE
Add typehints ciphers and bool alg

### PR DESCRIPTION
**Pull request by jenia90**
#2128 #2510 Added missing type hints to cipher and and bool alg python files.

* [ ] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: boolean_algebra/quine_mc_cluskey.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def check(binary):
         binary = list(set(temp))
 
 
-def decimal_to_binary(no_of_variable, minterms):
+def decimal_to_binary(no_of_variable: int, minterms: [int]) -> [str]:
```
Comment (poyea) [inline boolean_algebra/quine_mc_cluskey.py:46]: Is it `float`?

File: ciphers/decrypt_caesar_with_chi_squared.py [hunk 2]
```diff
@@ -3,10 +3,10 @@
 
 def decrypt_caesar_with_chi_squared(
     ciphertext: str,
-    cipher_alphabet=None,
-    frequencies_dict=None,
+    cipher_alphabet: str = None,
+    frequencies_dict: str = None,
     case_sensetive: bool = False,
-) -> tuple:
+) -> (int, float, str):
```
Comment (poyea) [inline ciphers/decrypt_caesar_with_chi_squared.py:9]: I'm not sure will `Tuple[int, float, str]` work?

File: ciphers/rsa_key_generator.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ def main():
     print("Key files generation successful.")
 
 
-def generateKey(keySize):
+def generateKey(keySize: int) -> (tuple, tuple):
```
Comment (poyea) [inline ciphers/rsa_key_generator.py:15]: Or `tuple[int, int]`?

**Review by poyea — COMMENTED** [review]
I've made some comments for you. Please re-check some of them (maybe I made a mistake) before we have this landed. Very nice work!! Thank you for your pull request!🤩

File: ciphers/rsa_key_generator.py (hunk 3 above)
Comment (jenia90) [reply ciphers/rsa_key_generator.py:15]: yes, probably should write it more explicitly .

File: ciphers/decrypt_caesar_with_chi_squared.py (hunk 2 above)
Comment (jenia90) [reply ciphers/decrypt_caesar_with_chi_squared.py:9]: You can write it both ways, but I'll make the change and commit the relevant changes.

**Comment by jenia90** [comment]
@poyea any suggestion how I can fix rsa_key_generator.py not being able to import modules from ciphers folder? It seems like this is the only thing stopping pre-commit from passing, however I can't see any problem with the way it is currently written.

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Pull request merged**